executor/src/witgen/machines/block_machine.rs

```diff
@@ -561,6 +561,11 @@ impl<'a, T: FieldElement> BlockMachine<'a, T> {
                 );
                 self.append_block(new_block)?;
 
+                // TODO: This would be the right thing to do, but currently leads to failing tests
+                // due to #1385 ("Witgen: Block machines "forget" that they already completed a block"):
+                // https://github.com/powdr-labs/powdr/issues/1385
+                // let updates = updates.report_side_effect();
```

Comment:
`test_data/asm/book/operations_and_links.asm` fails, because there is a machine that calls the same sub-machine with multiple arguments. Because of #1385, this keeps adding new blocks. Not reporting the side effect makes this works, because then the identity is not processed again.

Reply:
is it gonna be fixed here or later?

Reply:
Later, this requires fixing #1385, which is not the scope of this PR. The fix in this PR is disabled for block machines because of #1385, but IMO it makes sense to merge like this because it adds all the infrastructure to handle side effects, and it already works for other machines (crucially: memory).